src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// An <see cref="IHtmlLocalizer"/> that uses the <see cref="IStringLocalizer"/> to provide localized HTML content.
+    /// </summary>
+    public class HtmlLocalizer : IHtmlLocalizer
+    {
+        private readonly IStringLocalizer _localizer;
+        private readonly IHtmlEncoder _encoder;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/> to read strings from.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizer localizer, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizer = localizer;
+            _encoder = encoder;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="HtmlLocalizer"/>.</returns>
+        public virtual IHtmlLocalizer WithCulture([NotNull] CultureInfo culture) =>
```

Comment:
Why so many `virtual` methods and properties in this PR? The implementations are mostly one-liners and the classes can be replaced in DI.